service/src/main/kotlin/fi/espoo/evaka/reports/VardaErrorsReport.kt

```diff
@@ -58,17 +58,35 @@ FROM varda_service_need vsn
 JOIN service_need sn on vsn.evaka_service_need_id = sn.id
 JOIN service_need_option sno ON sn.option_id = sno.id
 LEFT JOIN varda_reset_child vrc ON vrc.evaka_child_id = vsn.evaka_child_id
-WHERE vsn.update_failed = true AND vrc.reset_timestamp IS NOT NULL
-ORDER BY vsn.updated DESC
+WHERE
+    vsn.update_failed AND
+    vrc.reset_timestamp IS NOT NULL AND
+    NOT EXISTS (SELECT FROM varda_state vs WHERE vs.child_id = vsn.evaka_child_id)
+
+UNION ALL
+
+SELECT
+    NULL AS service_need_id,
+    NULL AS service_need_validity,
+    NULL AS service_need_option_name,
+    child_id,
+    errored_at AS updated,
+    coalesce(last_success_at, created_at) AS created,
+    ARRAY[error] AS errors,
+    NULL AS reset_timestamp
+FROM varda_state
+WHERE error IS NOT NULL
```

Comment:
partial index was for errored_at so should maybe use that here too

Reply:
Good catch, fixed.